### PR TITLE
Move common parts to a separate ubuntu_wizard package

### PR DESCRIPTION
**Pull request by jpnurmi**
Will be shared with the WSL setup (#160).

### CODE REVIEW

**Comment by jpnurmi** [comment]
https://github.com/canonical/ubuntu-desktop-installer/pull/232 helps with the formatting issue

**Comment by oSoMoN** [comment]
How are localizations in this new package handled?
I modified `packages/ubuntu_wizard/lib/src/l10n/ubuntu_fr_FR.arb` to add translations for the password strength related strings, but `packages/ubuntu_wizard/lib/src/l10n/ubuntu_localizations_fr.dart` wasn't re-generated, and as a consequence my translations weren't picked up in the UI.

**Comment by jpnurmi** [comment]
> How are localizations in this new package handled?

Sorry, I forgot to explain this. In short: 1) edit .arb, 2) run flutter gen-l10n, and 3) commit lib/src/l10n.

Unlike ubuntu_desktop_installer's localizations, ubuntu_wizard's localizations are not auto-generated into a hidden build directory. The localizations must be available/exported from the package to the app, so they are generated into the source tree.

I wonder if we could actually do this for the app too because the auto-generation is either somehow unreliable or Dart-Code has problems resolving the generated files (need to run pub get multiple times). Have you had that problem too?

File: packages/ubuntu_wizard/linux/ubuntu_wizard_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+static void on_method_call(FlMethodChannel* channel, FlMethodCall* method_call,
+                           gpointer user_data) {
+  FlPluginRegistrar* registrar = FL_PLUGIN_REGISTRAR(user_data);
+  ubuntu_wizard_plugin_handle_method_call(registrar, method_call);
+}
+
+static gboolean on_delete_event(GtkWidget* /*window*/, GdkEvent* /*event*/,
+                                gpointer user_data) {
+  g_autoptr(GError) error = nullptr;
+  FlValue* event = fl_value_new_string("deleteEvent");
```
Comment (robert-ancell) [inline packages/ubuntu_wizard/linux/ubuntu_wizard_plugin.cc:64]: This should be `g_autoptr(FlValue) event = fl_value_new_string("deleteEvent")`.

File: packages/ubuntu_wizard/lib/src/l10n/ubuntu_nl_NL.arb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "@@locale": "nl_NL"
+}
```
Comment (robert-ancell) [inline packages/ubuntu_wizard/lib/src/l10n/ubuntu_nl_NL.arb:3]: Nit: A number of these translation files don't have trailing newlines.

File: packages/ubuntu_wizard/lib/src/l10n/ubuntu_localizations.dart [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+/// dependencies:
+///   # Internationalization support.
+///   flutter_localizations:
+///     sdk: flutter
+///   intl: any # Use the pinned version from flutter_localizations
+///
+///   # rest of dependencies
+/// ```
+///
+/// ## iOS Applications
```
Comment (robert-ancell) [inline packages/ubuntu_wizard/lib/src/l10n/ubuntu_localizations.dart:50]: Is this copy-pasted from somewhere?

File: packages/ubuntu_wizard/linux/ubuntu_wizard_plugin.cc (hunk 1 above)
Comment (robert-ancell) [reply packages/ubuntu_wizard/linux/ubuntu_wizard_plugin.cc:64]: (This is a bug in the existing code).

**Review by robert-ancell — APPROVED** [review]
Other than the comments given, LGTM.

File: packages/ubuntu_wizard/lib/src/l10n/ubuntu_localizations.dart (hunk 3 above)
Comment (jpnurmi) [reply packages/ubuntu_wizard/lib/src/l10n/ubuntu_localizations.dart:50]: auto-generated by `flutter gen-l10n`: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart#L54

**Comment by oSoMoN** [comment]
> > How are localizations in this new package handled?
> 
> Sorry, I forgot to explain this. In short: 1) edit .arb, 2) run flutter gen-l10n, and 3) commit lib/src/l10n.

Ack. This should be documented in the "Translations" section of CONTRIBUTING.md.


> Unlike ubuntu_desktop_installer's localizations, ubuntu_wizard's localizations are not auto-generated into a hidden build directory. The localizations must be available/exported from the package to the app, so they are generated into the source tree.
> 
> I wonder if we could actually do this for the app too because the auto-generation is either somehow unreliable or Dart-Code has problems resolving the generated files (need to run pub get multiple times). Have you had that problem too?

Yes, I see that from time to time. I have no strong opinion on this, we can try versioning the generated localizations in the source tree and see how (in)convenient it is.

**Comment by jpnurmi** [comment]
> Yes, I see that from time to time. I have no strong opinion on this, we can try versioning the generated localizations in the source tree and see how (in)convenient it is.

Yeah, it's easy to forget to update. What do you think about having a CI check that re-generates all generated files and checks if there are any changes?

**Comment by oSoMoN** [comment]
> > Yes, I see that from time to time. I have no strong opinion on this, we can try versioning the generated localizations in the source tree and see how (in)convenient it is.
> 
> Yeah, it's easy to forget to update. What do you think about having a CI check that re-generates all generated files and checks if there are any changes?

Good idea!

**Pull request merged**